src/lib.rs

```diff
@@ -184,6 +183,38 @@ impl<'de, T: Deserialize<'de>> Deserialize<'de> for Secret<T> {
     }
 }
 
+#[cfg(feature = "serde")]
+#[derive(Serialize)]
+struct Test {
+    #[serde(serialize_with = "expose_secret")]
+    one: Secret<String>,
+    #[serde(serialize_with = "expose_secret")]
+    two: Option<Secret<String>>,
+}
+
+pub trait ContainsSecret<T> {
+    type Exposed<'a>: Serialize
+    where
+        Self: 'a;
+    fn expose_secret<'a>(&'a self, expose: impl Fn(&Secret<T>) -> &T) -> Self::Exposed<'a>;
```

Comment:
I ask the trait user here for the `expose_secret` function here because I want that function to be the only way to expose secrets.